indexer-app/src/cmr/indexer/data/concepts/collection.clj

```diff
@@ -253,7 +260,11 @@
             :user-id user-id
             :permitted-group-ids permitted-group-ids
             ;; If there's an entry in the collection granule aggregates then the collection has granules.
-            :has-granules (some? (cgac/get-coll-gran-aggregates context concept-id))
+            :has-granules has-granules
+            :has-granules-or-cwic (or
```

Comment:
Put the simpler clause first to short-circuit an or.
```
(or has-granules
    (some #(= (cwic-tag) %) (map :tag-key.lowercase tags)))
```

Reply:
the second returns nil, which breaks the sort.  You don't end up having a field there. I tried this at first.  Had to reverse the order to get it to work.

Reply:
I go with `(some? (some #(....` when it needs to be a boolean.

Reply:
I didn't want to do `(some? (some ...) ...)` because my instinct told me it probably wasn't idiomatic.

Reply:
We've actually discussed it in-channel several times, and for cases when you must have a true or false, not just true or nil, we've agreed to go this route. 

(Tbh, we should code more to Clojure's treatment of nil, and not make demands as if it was a strongly typed language, but that decision was made long before I showed up! ;-))